consensus/src/backup/saver.rs

```diff
@@ -100,3 +99,133 @@ impl<H: Hasher, D: Data, MK: MultiKeychain, W: Write> BackupSaver<H, D, MK, W> {
+        NodeCount, NodeIndex, Terminator,
+    };
+
+    type TestBackupSaver = BackupSaver<Hasher64, Data, Keychain, Saver>;
+    type TestUnit = UncheckedSignedUnit<Hasher64, Data, Signature>;
+    type TestAlertData = AlertData<Hasher64, Data, Keychain>;
+    struct PrepareSaverResponse<F: futures::Future> {
+        task: F,
+        units_for_saver: mpsc::UnboundedSender<TestUnit>,
+        units_from_saver: mpsc::UnboundedReceiver<TestUnit>,
+        alerts_for_saver: mpsc::UnboundedSender<TestAlertData>,
+        alerts_from_saver: mpsc::UnboundedReceiver<TestAlertData>,
+        exit_tx: oneshot::Sender<()>,
+    }
+
+    fn prepare_saver() -> PrepareSaverResponse<impl futures::Future> {
+        let (units_for_saver, units_from_runway) = mpsc::unbounded();
+        let (units_for_runway, units_from_saver) = mpsc::unbounded();
+        let (alerts_for_saver, alerts_from_alerter) = mpsc::unbounded();
+        let (alerts_for_alerter, alerts_from_saver) = mpsc::unbounded();
+        let (exit_tx, exit_rx) = oneshot::channel();
+        let backup = Saver::new();
+
+        let task = {
+            let mut saver: TestBackupSaver = BackupSaver::new(
+                units_from_runway,
+                alerts_from_alerter,
+                units_for_runway,
+                alerts_for_alerter,
+                backup,
+            );
+
+            async move {
+                saver.run(Terminator::create_root(exit_rx, "saver")).await;
+            }
+        };
+
+        PrepareSaverResponse {
+            task,
+            units_for_saver,
+            units_from_saver,
+            alerts_for_saver,
+            alerts_from_saver,
+            exit_tx,
+        }
+    }
+
+    #[tokio::test]
+    async fn test_proper_response_ordering() {
+        let PrepareSaverResponse {
+            task,
+            units_for_saver,
+            mut units_from_saver,
+            alerts_for_saver,
+            mut alerts_from_saver,
+            exit_tx,
+        } = prepare_saver();
+
+        let handle = tokio::spawn(async {
+            task.await;
+        });
+
+        let creators = creator_set(NodeCount(5));
+        let keychains: Vec<_> = (0..5)
+            .map(|id| Keychain::new(NodeCount(5), NodeIndex(id)))
+            .collect();
+        let units: Vec<TestUnit> = (0..5)
+            .map(|k| {
+                preunit_to_unchecked_signed_unit(
+                    creators[k].create_unit(0).unwrap().0,
+                    0,
+                    &keychains[k],
+                )
+            })
+            .collect();
+        let alerts: Vec<TestAlertData> = (0..5)
+            .map(|k| {
+                TestAlertData::OwnAlert(Alert::new(
+                    NodeIndex(0),
+                    (units[k].clone(), units[k].clone()),
+                    vec![],
+                ))
+            })
+            .collect();
+
+        let backup_save_ordering = vec![0, 1, 1, 1, 0, 0, 1, 0, 1, 0];
```

Comment:
nit: instead of 0,1 use enum { Alert(alert), Unit(unit) } .
the nyou can iterate this vector and get rid of i, UI and ai.

Reply:
I was rather thinking of smth like this:
```
let backup_save_ordering = vec![Alert(alerts[0].clone), Unit(units[0.clone()]), ...];
```
but feel free to resolve this

Reply:
then we need to index them all manually which I don't like tbh :P